pkg/config/config.go

```diff
@@ -32,6 +32,22 @@ type ComponentSettings struct {
 	// The builder image to use
 	ComponentType *string `yaml:"ComponentType,omitempty"`
 
+	// Path is path to binary in current/context dir
+	Path *string `yaml:"Path,omitempty"`
+
+	// Ref is component source git ref but can be levaraged for more in future
+	Ref *string `yaml:"Ref,omitempty"`
+
+	// Type is type of component source: git/local/binary
+	SourceType *string `yaml:"SourceType,omitempty"`
+
+	// Ports is a slice of ports to be exposed when a component is created
+	Ports *string `yaml:"Ports,omitempty"`
```

Comment:
slice like we discussed on call with @kadel ?

Reply:
done